src/lib/components/dropdown/Dropdown.component.js

```diff
@@ -101,86 +99,125 @@ const Caret = styled.span`
 
 const TriggerStyled = ButtonStyled.withComponent("div");
 
+const DropdownTriggerContainer = forwardRef<DropdownTriggerContainerProps, Element>(({isItem, dataIndex, open, size , variant, title, onBlur, onFocus, onClick, onMouseEnter, onMouseLeave, children, ...rest}, ref) => {
+  return isItem ? (
+    <DropdownMenuItemStyled
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
+      data-index={dataIndex}
+      onClick={onClick}
+      ref={ref}
+    >
+      {children}
+    </DropdownMenuItemStyled>
+  ) : (
+    <DropdownStyled
+      active={open}
+      variant={variant}
+      className="sc-dropdown"
+      {...rest}
+    >
+    <TriggerStyled
+      variant={variant}
+      size={size}
+      className="trigger"
+      onBlur={onBlur}
+      onFocus={onFocus}
+      onClick={onClick}
+      tabIndex="0"
+      title={title}
+      ref={ref}
+    >
+      {children}
+    </TriggerStyled>
+    </DropdownStyled>
+  )
+})
+
 function Dropdown({
+  isItem = false,
   items,
   text,
   icon,
   size = "base",
   variant = "base",
   title,
   caret = true,
+  dataIndex = null,
+  onClick = null,
   ...rest
 }: Props) {
   const [open, setOpen] = useState(false);
   const [menuSize, setMenuSize] = useState();
   const [triggerSize, setTriggerSize] = useState();
+  const [itemIndex, setItemIndex] = useState()
 
   const refMenuCallback = useCallback(node => {
     if (node !== null) {
       setMenuSize(node.getBoundingClientRect());
     }
-  }, []);
+  }, [setMenuSize]);
 
   const refTriggerCallback = useCallback(node => {
     if (node !== null) {
       setTriggerSize(node.getBoundingClientRect());
     }
-  }, []);
+  }, [setTriggerSize]);
 
   return (
-    <DropdownStyled
-      active={open}
-      variant={variant}
-      className="sc-dropdown"
-      {...rest}
-    >
-      <TriggerStyled
+      <DropdownTriggerContainer
+        open={open}
+        isItem={isItem}
+        onMouseEnter={(e) => {
+          setItemIndex(e && e.target && e.target.getAttribute('data-index') || 0)
```

Comment:
We got a flow error. https://eslint.org/docs/rules/no-mixed-operators
```setItemIndex(e && e.target && e.target.getAttribute('data-index')) || 0```

Reply:
done